lightning/src/ln/channelmanager.rs

```diff
@@ -6179,7 +6210,8 @@ where
 	}
 
 	fn block_disconnected(&self, header: &BlockHeader, height: u32) {
-		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(&self.total_consistency_lock, &self.persistence_notifier);
+		let _persistence_guard = PersistenceNotifierGuard::optionally_notify(&self.total_consistency_lock,
+			&self.persistence_notifier, || -> NotifyOption { NotifyOption::DoPersist });
```

Comment:
Are we doing this to avoid calling `process_background_events`? Not sure why we're doing that

Reply:
I don't see why :( could you elaborate?

Reply:
Oh, I found it: https://github.com/lightningdevkit/rust-lightning/pull/2167/files#diff-645d12c7269d09caab57d2e0c1fd34f672ee259994c86ad4bad7f4da9183671cR6418 basically the first comment in `do_chain_event`.

Reply:
I think some more docs on `pending_background_events` and/or `process_background_events` would be helpful